### PR TITLE
M3 oauth fix

### PR DESCRIPTION
**Pull request by escopecz**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Automated tests included? | /
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8024
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

Creating API credentials nor API access via Oauth1a did not work on the M3 branch. This PR fixes that. 

When I was in it, I tested also Oauth2 and it had similar issues so this PR fixes those too.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Try to create Oauth1 credentials
2. You won't be able to do that due to various errors.

#### Steps to test this PR:
1. Try now. You'll be able to create the credentials now. 
2. And you'll be also able to authenticate against Mautic with Oauth1 and make successful API requests. I used [API tester](https://github.com/mautic/api-library) to test that.
3. Test Oauth2 too. Clear session in the API tester for that.

### CODE REVIEW

File: app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php [hunk 1]
```diff
@@ -12,64 +12,138 @@
...
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    /**
+     * This constructor must be duplicated from the extended class so our custom code could access the properties.
+     */
+    public function __construct(
```
[the remark this answers is not in the document]
Comment (escopecz) [reply app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php]: We decided that with PHP 7.2+ the param and return type comments doesn't make sense.

File: app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php [hunk 2]
```diff
@@ -12,64 +12,138 @@
...
+        Form $authorizeForm,
+        AuthorizeFormHandler $authorizeFormHandler,
+        OAuth2 $oAuth2Server,
+        EngineInterface $templating,
+        TokenStorageInterface $tokenStorage,
+        UrlGeneratorInterface $router,
+        ClientManagerInterface $clientManager,
+        EventDispatcherInterface $eventDispatcher,
+        SessionInterface $session = null,
+        $templateEngineType = 'php'
```
[the remark this answers is not in the document]
Comment (escopecz) [reply app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php]: I see added value in this. It names what the 'php' string means.

[the remark this answers is not in the document]
Comment (hluchas) [reply app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php]: > I see added value in this. It names what the 'php' string means.

But I noticed that default value definition in constructor is not used, thus not needed.

[the remark this answers is not in the document]
Comment (escopecz) [reply app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php]: What change do you propose exactly?

[the remark this answers is not in the document]
Comment (hluchas) [reply app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php]: Remove this line.

[the remark this answers is not in the document]
Comment (escopecz) [reply app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php]: What do you want to do with line 101 then?

[the remark this answers is not in the document]
Comment (hluchas) [reply app/bundles/ApiBundle/Controller/oAuth2/AuthorizeController.php]: Leave it a it is.

**Comment by escopecz** [comment]
@anton-vlasenko could you please test this when you get a chance? I think that Lukas' issues are on his local. It works on mine.

**Pull request merged**